src/features/foods/components/FilterButton.tsx

```diff
@@ -108,93 +123,109 @@ export const FilterButton = ({
    * Process when group filter button is pushed
    * @param event
    */
-  const handleSelectGroup = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    setSelectedGroup(event.target.value);
+  const handleSelectGroup = (value: string) => {
+    setSelectedGroup(value);
+    setSelectedContainer('');
   };
 
   /**
    * Process when container filter button is pushed
    * @param event
    */
-  const handleSelectContainer = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    setSelectedContainer(event.target.value);
+  const handleSelectContainer = (containerId: string) => {
+    setSelectedContainer(containerId);
+    setSelectedGroup(containerIdGroupIdMap[containerId]);
   };
   const [isDrawerOpen, setIsDrawerOpen] = useState(false);
```

Comment:
For better readability, please consider grouping all useState declarations together. 
For instance, moving the one at line 139 up to line 58 would enhance code cohesion.

Feel free to adjust as per your preference tho!

Reply:
Thanks! Updated!